.github/workflows/build-website.yml

```
@@ -0,0 +1,44 @@
+name: build-website 
+
+on:
+  push:
+    branches: [ docs ]
+
+jobs:
+  build-linux:
+    runs-on: ubuntu-latest
+    strategy:
+      max-parallel: 1
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+
+    - name: Install dependencies
+      run: |
+        sudo apt-get install python3-sphinx
+        python3 -m pip install nbsphinx
+        python3 -m pip install sphinx_rtd_theme
+        python3 -m pip install sphinx_book_theme
+        python3 -m pip install sphinx-toolbox --user
+        python3 -m pip install sphinx-jinja2-compat==0.2.0b1
+        python3 -m pip install sphinx-autoapi
+
+    - name: make the sphinx docs
+      run: |
+        make -C docs clean 
+        # sphinx-apidoc -f -o docs/source . -H Test -e -t docs/source/_templates
+        make -C docs html
+
+    - name: Deploy
+      uses: peaceiris/actions-gh-pages@v3
+      # If you're changing the branch from main, 
+      # also change the `main` in `refs/heads/main` 
+      # below accordingly.
+      if: ${{ github.ref == 'refs/heads/docs' }}
+      with:
+        github_token: ${{ secrets.GITHUB_TOKEN }}
+        publish_dir: ./docs/build/html
```

```suggestion
        publish_dir: ./docs/build/html
        commit_message: "update docs at commit: ${{ github.event.head_commit.message }}"
```
Might be a good idea to add a commit so you can track the progression of the `gh-pages` branch in the source tree.